### PR TITLE
[FEATURE] Support build minification excludes

### PR DESCRIPTION
**Pull request by larskissel**
Supports the build configuration "minification excludes"
introduced in specVersion 2.6. This allows to exclude
JavaScript resources from minification (tasks: uglify and
createDebugFiles).

JIRA: CPOUI5FOUNDATION-405

### CODE REVIEW

**Comment by larskissel** [comment]
Integration tests will be uploaded with the next commit.

File: lib/types/application/ApplicationBuilder.js [hunk 1]
```diff
@@ -138,12 +138,17 @@ class ApplicationBuilder extends AbstractBuilder {
 			});
 		}
 
+		const minificationPattern = ["/**/*.js"];
+		if (["2.6"].includes(project.specVersion)) {
+			const minification = (project.builder && project.builder.minification) || {};
+			this.enhancePatternWithExcludes(minificationPattern, minification.excludes);
```
Comment (matz3) [inline lib/types/application/ApplicationBuilder.js:144]: Should the pattern be prefixed with `/resources/` here?
The default pattern for applications is `/**/*.js`, so I'd expect the excludes to be possible for all paths, not just within `/resources/`.
Or do I missing something here? For libraries it totally makese sense as we only process `/resources/`.

**Review by matz3 — CHANGES_REQUESTED** [review]

File: test/lib/builder/builder.js [hunk 2]
```diff
@@ -1614,14 +1690,50 @@ const libraryJTree = {
 		"configuration": {
 			"paths": {
 				"src": "main/src"
-			}
+			},
+			"propertiesFileSourceEncoding": "ISO-8859-1"
```
Comment (RandomByte) [inline test/lib/builder/builder.js:1694]: Why this change?

Comment (larskissel) [reply test/lib/builder/builder.js]: Looks like I made copy&paste to the wrong object. Done.
